sources/api/apiserver/src/server/controller.rs

```diff
@@ -443,19 +591,161 @@ pub(crate) fn get_metadata_for_all_data_keys<D: DataStore, S: AsRef<str>>(
+{
+    // Get settings to commit from pending transaction
+    let settings_to_commit = datastore
+        .get_prefix("settings.", committed)
+        .context(error::DataStoreSnafu { op: "get_prefix" })?;
+
+    // Get metadata from pending transaction
+    let mut transaction_metadata = datastore
+        .get_metadata_prefix("settings.", committed, &None as &Option<&str>)
+        .context(error::DataStoreSnafu {
+            op: "get_metadata_prefix",
+        })?;
+
+    let mut metadata_to_commit: Vec<(Key, Key, String)> = Vec::new();
+
+    // Parse and validate all the metadata enteries from pending transaction
+    for (key, value) in transaction_metadata.iter_mut() {
+        for (metadata_key, metadata_value) in value {
+            // For now we are only processing the strength metadata from pending
+            // transaction to live
+            if metadata_key.name() != "strength" {
+                warn!(
+                    "Metadata key is {}, Skipping the commit as we only allow Strength metadata.",
+                    metadata_key.name()
+                );
+                continue;
+            }
+
+            // strength in pending transaction
+            let pending_strength: Strength =
+                deserialize_scalar::<String, ScalarError>(metadata_value.as_ref())
+                    .with_context(|_| error::DeserializeStrengthSnafu {})?
+                    .parse::<Strength>()
+                    .context(error::ParseStrengthSnafu)?;
+
+            // Get the setting strength in live
+            // get_metadata function returns Ok(None) in case strength does not exist
+            // We will consider this case as strength equals strong.
+            let committed_strength: Strength = datastore
+                .get_metadata(metadata_key, key, &Committed::Live)
+                .context(error::DataStoreSnafu { op: "get_metadata" })?
+                .map(|x| x.parse::<Strength>())
+                .transpose()
+                .context(error::ParseStrengthSnafu)?
+                .unwrap_or_default();
+
+            let may_be_value = datastore
+                .get_key(key, &Committed::Live)
+                .context(error::DataStoreSnafu { op: "get_key" })?;
+
+            trace!(
+                "datastore_transaction_check: key: {:?}, metadata_key: {:?}, metadata_value: {:?}",
+                key.name(),
+                metadata_key.name(),
+                metadata_value
+            );
+
+            match (pending_strength, committed_strength, may_be_value) {
+                (Strength::Weak, Strength::Strong, None)
+                | (Strength::Strong, Strength::Weak, Some(..))
+                | (Strength::Strong, Strength::Weak, None) => {
+                    let met_value = serialize_scalar::<_, ScalarError>(&pending_strength)
+                        .with_context(|_| error::SerializeSnafu {})?;
+
+                    metadata_to_commit.push((metadata_key.clone(), key.clone(), met_value));
```

Comment:
Does this result in actually committing "strength: Strong" metadata?

Reply:
This will only write when the strength=weak already exist. Otherwise, it will create weak strength file.